aggregator/src/aggregator.rs

```diff
@@ -661,38 +661,29 @@ impl<C: Clock> Aggregator<C> {
                     Error::InvalidTask(*task_id, OptOutReason::InvalidParameter(err.to_string()))
                 })?;
 
-        let our_role = match peer_role {
-            Role::Leader => Role::Helper,
-            Role::Helper => Role::Leader,
-            _ => {
-                return Err(Error::Internal(
-                    "role should have only been Helper or Leader".to_string(),
-                ))
-            }
-        };
-
         let vdaf_verify_key = peer_aggregator.derive_vdaf_verify_key(task_id, &vdaf_instance);
 
-        let task = taskprov::Task::new(
+        let task = AggregatorTask::new(
             *task_id,
             leader_url,
-            helper_url,
             task_config.query_config().query().try_into()?,
             vdaf_instance,
-            our_role,
             vdaf_verify_key,
             task_config.query_config().max_batch_query_count() as u64,
             Some(*task_config.task_expiration()),
             peer_aggregator.report_expiry_age().cloned(),
             task_config.query_config().min_batch_size() as u64,
             *task_config.query_config().time_precision(),
             *peer_aggregator.tolerable_clock_skew(),
+            // Taskprov task has no per-task HPKE keys
+            [],
```

Comment:
It seems then the HPKE keys should be part of the (normal) Helper and Leader variants of AggregatorTaskParameters. I'm generally not a fan of passing false data (although I suppose an empty slice is not really _false_ per se). WDYT?

Reply:
I tried that for the reasons you say. The issue is that then you can't have `AggregatorTask::hpke_keys(&self) -> &HashMap<HpkeConfigId, HpkeKeypair>`, because not all variants have HPKE keys. From there there are two options:

- return `Option<&HashMap<HpkeConfigId, HpkeKeypair>>` and return `None` for task prov tasks. This forces the caller to do annoying `Option` unwrapping or iterator flattening gymnastics despite `None` not being any more expressive than an empty hash.
- return an owned `HashMap<...>` value and return an empty hash for the taskprov case. This is unfortunate because it forces an allocation on one path, or a clone (and hence allocation) on the other (you can't `return &HashMap::new()` because the value wouldn't live long enough).

All three options are kind of awkward, but I think this is the least awkward. I do think we can do better here, by further unifying the taskprov and non-taskprov cases such that code holding an `AggregatorTask` value doesn't need to know where the task came from. `AggregatorTask::hpke_keys` would yield a map of keys, which would have been constructed from the `global_hpke_keys` table in the taskprov case or `task_hpke_keys` otherwise. But I don't propose to do that refactoring now.